src/events/types/inventory/PrepareItemEnchantEvent.ts

```diff
@@ -0,0 +1,66 @@
+import { InventoryEvent } from "./InventoryEvent";
+import { Player } from "../../../player/Player";
+import { Block } from "../../../block/Block";
+import { Inventory } from "../../../inventory/Inventory";
+import { EnchantingInventory } from "../../../inventory/types";
+import { ItemStack } from "../../../material/ItemStack";
+import { Cancellable } from "../../Cancellable";
+import { EnchantmentOffer } from "../../../material/EnchantmentOffer";
+
+export class PrepareItemEnchantEvent extends InventoryEvent implements Cancellable {
+
+  public static getBukkitClasspath(): string {
+    return 'org.bukkit.event.inventory.PrepareItemEnchantEvent';
+  }
+
+  getInventory(): EnchantingInventory {
+    return Inventory.fromJava<EnchantingInventory>(this.toJava().getInventory());
+  }
+
+  /**
+   * Gets the player enchanting the item
+   */
+  getEnchanter(): Player {
+    return Player.fromJava(this.toJava().getEnchanter());
+  }
+
+  /**
+   * Gets the block being used to enchant the item
+   */
+  getEnchantBlock(): Block {
+    return Block.fromJava(this.toJava().getEnchantBlock());
+  }
+
+  /**
+   * Gets the item to be enchanted.
+   */
+  getItem(): ItemStack {
+    return ItemStack.fromJava(this.toJava().getItem());
+  }
+
+  /**
+   * Get a list of available EnchantmentOffer for the player. You can modify
+   * the values to change the available offers for the player. An offer may be
+   * null, if there isn't a enchantment offer at a specific slot. There are 3
+   * slots in the enchantment table available to modify.
+   */
+  getOffers(): EnchantmentOffer[] {
+    return this.toJava().getOffers().map((offer: Java.Value) => offer && EnchantmentOffer.fromJava(offer));
+  }
```

Comment:
If the offer value is nullable, the return type for this method should be changed to `(EnchantmentOffer | null)[]`.

Also the doc mentions that (in a normal Java plugin) you can edit the array to give the player different offers. I don't think this would work in JavaScript though.

Reply:
Correct about the type mismatch, and right about setting elements in the array wouldn't modify the values in java, I'll work on something to make that work. Otherwise, editing the EnchantmentOffers it gives you would modify those offers.

Reply:
Right right, just not editing the array itself. No worries, we'll add that functionality in another PR!